src/libraries/Common/src/System/Security/Cryptography/DSASecurityTransforms.macOS.cs

```diff
@@ -0,0 +1,218 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Formats.Asn1;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography.Apple;
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography
+{
+    internal static partial class DSAImplementation
+    {
+        public sealed partial class DSASecurityTransforms : DSA
+        {
+            public override DSAParameters ExportParameters(bool includePrivateParameters)
+            {
+                // Apple requires all private keys to be exported encrypted, but since we're trying to export
```

Comment:
Is this comment still true with the new key API?  (or is DSA just unchanged?)

Reply:
macOS code remains unchanged on all imports/exports to avoid bigger changes and breaking things in S.S.C.X509Certificates.

It is still true for the old style CSSM keys. The new iOS style keys don't have this requirement. The SecItemExport API doesn't work on the iOS style keys with the exception of RSA.

Reply:
Also, the new APIs / iOS doesn't support DSA at all. I have alternative set of changes where `DSA.Create` returns `PlatformNotSupportedException` directly but it requires further changes to CryptoConfig, unit tests, etc. I am not sure what is the preferred pattern here.